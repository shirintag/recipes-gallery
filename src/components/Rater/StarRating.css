.star-rating {
  display: flex;
  width: 120px;
  height: 30px;
  flex-wrap: wrap;
  justify-content: space-around;
}

.star-container {
  cursor: pointer;
  width: 22px;
}

.star {
  cursor: pointer;
  width: 18px;
  height: 18px;
  background-color: rgba(201, 201, 201);
  -webkit-clip-path: polygon(
    50% 0%,
    63% 38%,
    100% 38%,
    69% 59%,
    82% 100%,
    50% 75%,
    18% 100%,
    31% 59%,
    0% 38%,
    37% 38%
  );
  clip-path: polygon(
    50% 0%,
    63% 38%,
    100% 38%,
    69% 59%,
    82% 100%,
    50% 75%,
    18% 100%,
    31% 59%,
    0% 38%,
    37% 38%
  );
}

.star.selected {
  background-color: #fcd200;
  box-shadow: 0px 0px 8px 1px #503838 inset;
}
