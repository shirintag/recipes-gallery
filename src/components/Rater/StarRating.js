import React, { useState } from "react";
import Star from "./Star";

import "./StarRating.css";

const StarRating = ({ totalStars }) => {
  const [starsSelected, selectStar] = useState(0);

  return (
    <div className="star-rating">
      {[...Array(totalStars)].map((n, i) => (
        <Star
          key={i}
          selected={i < starsSelected}
          onClick={() => selectStar(i + 1)}
        />
      ))}
    </div>
  );
};

export default StarRating;
