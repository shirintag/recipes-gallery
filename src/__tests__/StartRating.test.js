import React from "react";
import { render } from "@testing-library/react";
import StarRating from "components/Rater/StarRating";

describe("StarRating", () => {
  it("should render", () => {
    const { container } = render(<StarRating />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should change selected stars on click", () => {
    let { container } = render(<StarRating totalStars={5} />);
    container = container.firstChild;
    // Initial rating
    let stars = container.querySelectorAll(".star");
    let starsSelected = container.querySelectorAll(".star.selected");
    expect(stars.length).toEqual(5);
    expect(starsSelected.length).toEqual(0);

    // Rate 4 stars
    stars[3].click();
    starsSelected = container.querySelectorAll(".star.selected");
    expect(starsSelected.length).toEqual(4);
  });
});
