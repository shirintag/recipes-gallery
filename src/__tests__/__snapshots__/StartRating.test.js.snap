// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating should render 1`] = `
<div
  class="star-rating"
>
  <div
    class="star-container"
  >
    <div
      class="star"
    />
  </div>
</div>
`;
